### PR TITLE
Update dependencies and tighten/standardize constraints

### PR DESCRIPTION
**Pull request by parlough**
Overall, no functional changes, just general cleanup and maintenance to make the Flutter beta changes easier to review.

Updates some dependencies, makes dependency constraints tighter and consistent across samples, fixes some incorrect dependencies, and regenerates some generated files after updates.

### CODE REVIEW

**Comment by parlough** [comment]
Ignore the stage workflow failing. We expended the channel quota and it will take a few days after reducing expiration time for that to open up.

File: examples/cookbook/testing/widget/scrolling/pubspec.yaml [hunk 1]
```diff
@@ -7,13 +7,13 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  path_provider: ^2.0.1
-  path: ^1.8.0
+  path_provider: ^2.0.15
```
Comment (domesticmouse) [inline examples/cookbook/testing/widget/scrolling/pubspec.yaml:10]: Trailing white space

**Review by domesticmouse — APPROVED** [review]
Could some of this work be automated with dependabot?

**Comment by parlough** [comment]
Yes some of it could, good point! Not all examples are configured in dependabot yet. I'm going to be doing some reorganizing of the example code soon and will make sure to set up that configuration for all of them.

**Pull request merged**